crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/fetch/fetch_task.rs

```diff
@@ -0,0 +1,284 @@
+                                    fetch_pool.remove(&key);
+                                    continue;
+                                }
+                            }
+                        }
+
+                        if let Err(err) = internal_sender.fetch(key, space, source).await {
+                            match err {
+                                KitsuneP2pError::GhostError(GhostError::Disconnected) => {
+                                    tracing::warn!("Fetch task is shutting down because the internal sender is closed");
+                                    break 'task_loop;
+                                }
+                                // TODO are these so common that we can discard them? Should there be a metric to track this?
+                                _ => tracing::debug!(?err),
+                            }
+                        }
+                    }
+
+                    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                }
+
+                tracing::info!("Fetch task is finishing");
+                this.write().is_finished = true;
+            }
+        });
+
+        this
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::FetchTask;
+    use crate::spawn::actor::test_util::InternalStubTestSender;
+    use crate::spawn::actor::{Internal, KSpace};
+    use crate::spawn::test_util::{InternalStub, InternalStubTest};
+    use crate::HostStub;
+    use futures::FutureExt;
+    use ghost_actor::actor_builder::GhostActorBuilder;
+    use ghost_actor::{GhostControlSender, GhostSender};
+    use kitsune_p2p_fetch::test_utils::{test_key_hash, test_req_op, test_req_region, test_source};
+    use kitsune_p2p_fetch::FetchSource;
+    use kitsune_p2p_fetch::{FetchKey, FetchPool};
+    use kitsune_p2p_types::KOpHash;
+    use parking_lot::{Mutex, RwLock};
+    use std::collections::HashSet;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    #[tokio::test(start_paused = true)]
+    async fn fetch_single_op() {
+        let (_task, fetch_pool, internal_sender_test, held_op_data) =
+            setup(InternalStub::new()).await;
+
+        fetch_pool.push(test_req_op(1, None, test_source(1)));
+        wait_for_pool_n(&fetch_pool, 1).await;
+
+        let fetched = wait_for_fetch_n(internal_sender_test.clone(), 1).await;
+
+        // The item should get fetched
+        assert_eq!(1, fetched.iter().flatten().count());
+
+        // Simulate the requested item being sent back
+        held_op_data.lock().insert(test_key_hash(1));
+
+        // Move forwards by 5 minutes so that the item will be retried
+        tokio::time::advance(Duration::from_secs(5 * 60)).await;
+
+        // Then the item should be removed from the pool
+        wait_for_pool_n(&fetch_pool, 0).await;
+
+        internal_sender_test
+            .ghost_actor_shutdown_immediate()
+            .await
+            .unwrap();
+    }
+
+    #[ignore = "open question"]
```

Comment:
Unless I've missed something this case isn't covered and could go some way to explaining why Holochain is using so much data.

Reply:
I'm pretty sure the region thing is still a little half-baked. @maackle - do you remember are we actually even making region requests ever?

Reply:
Cool okay, yeah it looked unused but there were enough usages that I wasn't sure. I'll remove this test and follow up with a PR to remove the unused code because if it's not going to be needed then the code is easier to understand without it :) 

Thank you!